lib/matplotlib/artist.py

```diff
@@ -534,9 +534,14 @@ def pick(self, mouseevent):
                           mouseevent, self, **prop)._process()
 
         # Pick children
-        for a in self.get_children():
+        children = self.get_children()
+        for a in children:
             # make sure the event happened in the same Axes
             ax = getattr(a, 'axes', None)
+            # if subfigure must get the axes from it
+            if isinstance(a, mpl.figure.SubFigure):
+                children.extend(ax)
```

Comment:
This will add direct `Axes` children of the `SubFigure`, but they can also contain recursive `SubFigure`s that then contain `Axes` children, which I don't believe are caught here.

Reply:
Yes, you are right. I have made changes to instead add `subfigure` children, this would add all the `recursive subfigures`, and also handle the `events`.